Android.Jenkinsfile

```diff
@@ -0,0 +1,148 @@
+			environment {
+				APK_SIGN_ALIAS = "test.tutao.de"
+				APK_SIGN_KEY_PASS = credentials('apk-sign-key-pass')
+			}
+			when {
+				not { expression { params.PROD } }
+			}
+			agent {
+				label 'linux'
+			}
+			steps {
+				sh 'npm ci'
+				sh 'node android.js -b releaseTest test'
+				stash includes: "build/app-android/tutanota-${VERSION}-releaseTest.apk", name: 'apk'
+			}
+		}
+
+		stage('Publish android app: production') {
+			when {
+				expression { params.PROD }
+				expression { params.PUBLISH }
+			}
+			steps {
+				sh 'npm ci'
+
+				unstash 'apk'
+
+				script {
+					def filePath = "build/app-android/tutanota-${VERSION}-release.apk"
+					def tag = "tutanota-android-release-${VERSION}"
+
+					publishToNexus(groupId: "app",
+								   artifactId: "android",
+								   version: "${VERSION}",
+								   assetFilePath: "${WORKSPACE}/${filePath}")
+
+					androidApkUpload(googleCredentialsId: 'android-app-publisher-credentials',
+									 apkFilesPattern: "${filePath}",
+									 trackName: 'production',
+									 rolloutPercentage: '100%')
+
+					sh "git tag ${tag}"
+					sh "git push --tags"
+
+					withCredentials([string(credentialsId: 'github-access-token', variable: 'GITHUB_TOKEN')]) {
+						def checksum = sh(returnStdout: true, script: "sha256sum ${WORKSPACE}/${filePath}")
+						def releaseNotes = sh(
+							"""node buildSrc/createGithubReleasePage.js --name '${VERSION} (Android)' \
+																	 --milestone '${VERSION}' \
+																	 --tag '${tag}' \
+																	 --uploadFile '${WORKSPACE}/${filePath}' \
+																	 --platform android \
+																	 --apkChecksum ${checksum}"""
+						)
+					}
+				}
+			}
+		}
+
+		stage('Publish android app: staging') {
+			when {
+				not { expression { params.PROD } }
+				expression { params.PUBLISH }
+			}
+			steps {
+				unstash 'apk'
+
+				publishToNexus(groupId: "app",
+						artifactId: "android-test",
+						version: "${VERSION}",
+						assetFilePath: "${WORKSPACE}/build/app-android/tutanota-${VERSION}-releaseTest.apk")
+
+				// This doesn't publish to the main app on play store,
+				// instead it get's published to the hidden "tutanota-test" app
+				// this happens because the AppId is set to de.tutao.tutanota.test by the android build
+				// and play store knows which app to publish just based on the id
+				androidApkUpload(googleCredentialsId: 'android-app-publisher-credentials',
+						apkFilesPattern: "build/app-android/tutanota-${VERSION}-releaseTest.apk",
+						trackName: 'internal',
+						rolloutPercentage: '100%')
+			}
+		}
+	}
+}
+
+def publishToNexus(Map params) {
+	withCredentials([usernamePassword(credentialsId: 'nexus-publish', usernameVariable: 'NEXUS_USERNAME', passwordVariable: 'NEXUS_PASSWORD')]) {
+		sh  "curl --silent --show-error --fail " +
+			"-u '${NEXUS_USERNAME}':'${NEXUS_PASSWORD}' " +
+			"-X POST 'http://[fd:aa::70]:8081/nexus/service/rest/v1/components?repository=releases' " +
```

Comment:
Can we use domain name instead?

Reply:
nope, some issues with reverse proxy and whatnot, it won't work without some configuration changes on moritz. Can add a comment instead

Reply:
I'm more thinking if something is moved around but if it's a proxy problem I'm fine